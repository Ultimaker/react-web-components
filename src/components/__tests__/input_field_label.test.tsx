// Copyright (c) 2018 Ultimaker B.V.
import 'jest';
import 'jsdom-global/register';
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import InputFieldLabel from '../input_field_label';

describe('The InputFieldLabel component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: 'testInputField',
      type: 'text',
      label: 'Input label',
      labelLayoutWidth: '1/1',
      labelWidthBreakpoint: 'sm'
    };
    wrapper = shallow(<InputFieldLabel {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
