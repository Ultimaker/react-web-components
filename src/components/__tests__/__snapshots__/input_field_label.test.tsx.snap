// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The InputFieldLabel component should render 1`] = `
<div
  className="input-field__label layout__item u-1/1-sm"
>
  <div
    className="layout layout--gutter-sm"
  >
    <div
      className="layout__item u-fit"
    >
      <label
        htmlFor="testInputField"
      >
        Input label
      </label>
    </div>
  </div>
</div>
`;

exports[`The InputFieldLabel component should render info link 1`] = `
<div
  className="input-field__label layout__item u-1/1-sm"
>
  <div
    className="layout layout--gutter-sm"
  >
    <div
      className="layout__item u-fit"
    >
      <label
        htmlFor="testInputField"
      >
        Input label
      </label>
    </div>
    <div
      className="layout__item u-fit input-field__label-addition"
    >
      <InfoLink
        infoLinkURL="https://ultimaker.com/"
      />
    </div>
  </div>
</div>
`;

exports[`The InputFieldLabel component should render info tooltip 1`] = `
<div
  className="input-field__label layout__item u-1/1-sm"
>
  <div
    className="layout layout--gutter-sm"
  >
    <div
      className="layout__item u-fit"
    >
      <label
        htmlFor="testInputField"
      >
        Input label
      </label>
    </div>
    <div
      className="layout__item u-fit input-field__label-addition"
    >
      <InfoTooltip
        infoText="Test tooltip text"
      />
    </div>
  </div>
</div>
`;

exports[`The InputFieldLabel component should render pre label element before the label 1`] = `
<div
  className="input-field__label layout__item u-1/1-sm"
>
  <div
    className="layout layout--gutter-sm"
  >
    <div
      className="layout__item u-fit input-field__pre-element"
    >
      <div
        className="pre-label-element"
      />
    </div>
    <div
      className="layout__item u-fit"
    >
      <label
        htmlFor="testInputField"
      >
        Input label
      </label>
    </div>
  </div>
</div>
`;
